com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/scrbound.compute

```diff
@@ -1,538 +1,740 @@
+    float  bc; // Boundary coordinate with respect to the plane 'p'
+};
 
-    {
-        int f=0;
+ClipVertex CreateClipVertex(uint p, float4 v)
+{
+    bool evenPlane = (p & 1) == 0;
 
-        if(sideIndex==0 && lgtIndex<(int) g_iNrVisibLights)
-        {
-            // quick acceptance or rejection
-            unsigned int uCollectiveAnd = (unsigned int) -1;
-            unsigned int uCollectiveOr = 0;
-            for(f=0; f<6; f++)
-            {
-                unsigned int uFlagAnd = clipFlags[subLigt*6+f]&0x3f;
-                unsigned int uFlagOr = uFlagAnd;
-                for(int i=1; i<4; i++)
-                {
-                    unsigned int uClipBits = (clipFlags[subLigt*6+f]>>(i*6))&0x3f;
-                    uFlagAnd &= uClipBits;
-                    uFlagOr |= uClipBits;
-                }
+    float c = v[p >> 1];
+    float w = v.w;
 
-                uCollectiveAnd &= uFlagAnd;
-                uCollectiveOr |= uFlagOr;
-            }
+    ClipVertex cv;
 
-            bool bSetBoundYet = false;
-            float3 vMin=0.0, vMax=0.0;
-            if(uCollectiveAnd!=0 || uCollectiveOr==0)       // all invisible or all visible (early out)
-            {
-                if(uCollectiveOr==0)    // all visible
-                {
-                    for(f=0; f<6; f++)
-                    {
-                        const int sideIndex = f;
-
-                        float3 vFaceMi = float3(posX[subLigt*MAX_PNTS*2 + sideIndex + 0], posY[subLigt*MAX_PNTS*2 + sideIndex + 0], posZ[subLigt*MAX_PNTS*2 + sideIndex + 0]);
-                        float3 vFaceMa = float3(posX[subLigt*MAX_PNTS*2 + sideIndex + 6], posY[subLigt*MAX_PNTS*2 + sideIndex + 6], posZ[subLigt*MAX_PNTS*2 + sideIndex + 6]);
-
-                        for(int k=0; k<2; k++)
-                        {
-                            float3 vP = k==0 ? vFaceMi : vFaceMa;
-                            if(f==0 && k==0) { vMin=vP; vMax=vP; }
-
-                            vMax = max(vMax, vP); vMin = min(vMin, vP);
-                        }
-                    }
-                    bSetBoundYet=true;
-                }
-            }
-            else        // :( need true clipping
-            {
+    cv.pt = v;
+    cv.bc = evenPlane ? c : w - c; // dot(PlaneEquation, HapVertex);
 
-                for(f=0; f<6; f++)
-                {
-                    float3 q0, q1, q2, q3;
-                    GetHullQuad(q0, q1, q2, q3, boxX, boxY, boxZ, center, scaleXY, f);
+    return cv;
+}
 
-                    // 4 vertices to a quad of the convex hull in post projection space
-                    const float4 vP0 = mul(g_mProjection, float4(q0, 1));
-                    const float4 vP1 = mul(g_mProjection, float4(q1, 1));
-                    const float4 vP2 = mul(g_mProjection, float4(q2, 1));
-                    const float4 vP3 = mul(g_mProjection, float4(q3, 1));
+float4 IntersectEdgeAgainstPlane(ClipVertex v0, ClipVertex v1)
```

Comment:
Several function should be move to GeometricTool.hlsl

Reply:
Maybe in the future, if we end up reusing them.

Reply:
so ok ,but the problem is the reverse, people will never been aware that such a function exist, so they will duplicate it. But fair enough.